nvflare/lighter/impl/master_template.yml

```diff
@@ -475,3 +475,62 @@ docker_adm_sh: |
   DOCKER_IMAGE={~~docker_image~~}
   echo "Starting docker with $DOCKER_IMAGE"
   docker run --rm -it --name=fladmin -v $DIR/..:/workspace/ -w /workspace/ $DOCKER_IMAGE /bin/bash
+
+compose_yaml: |
+  services:
+    __overseer__:
+      build: ./nvflare
+      image: nvflare-service
+      volumes:
+        - .:/workspace
+      command: ["/workspace/startup/start.sh"]
+      ports:
+        - "8443:8443"
+
+    __flserver__:
+      image: nvflare-service
+      ports:
+        - "8002:8002"
+        - "8003:8003"
+      volumes:
+        - .:/workspace
+        - nvflare_svc_persist:/tmp/nvflare/
+      command: ["/usr/local/bin/python3",
+            "-u",
+            "-m",
+            "nvflare.private.fed.app.server.server_train",
+            "-m",
+            "/workspace",
+            "-s",
+            "fed_server.json",
+            "--set",
+            "secure_train=true",
+            "config_folder=config",
+          ]
+
+    __flclient__:
+      image: nvflare-service
+      volumes:
+        - .:/workspace
+      command: ["/usr/local/bin/python3",
+            "-u",
+            "-m",
+            "nvflare.private.fed.app.client.client_train",
+            "-m",
+            "/workspace",
+            "-s",
+            "fed_client.json",
+            "--set",
+            "secure_train=true",
+            "uid=__flclient__",
+            "config_folder=config",
+          ]
+
+  volumes:
+    nvflare_svc_persist:
+
+dockerfile: |
+  RUN pip install -U pip
+  RUN pip install nvflare
+  COPY requirements.txt requirements.txt
+  RUN pip install -r requirements.txt
```

Comment:
The updated project.yml config for DockerBuilder has
`requirements_file: docker_compose_requirements.txt`
which is copied into the build dir as requirements.txt, and then into the container image and installed in the dockerfile here.

Is this file required?  It seems like the docker build would fail on the COPY and RUN pip install -r requirements.txt if it is not supplied.  I'm thinking of a situation where the user is using a custom base image that already has their requirements installed, rather than the python3.8 base image.  We may want to add the logic to make this optional.

Reply:
Line 87 and below of docker.py handle that.
If requirements_file is not specified, default to requirements.txt.
If that file does not exist, an empty requirements.txt file is created in nvflare_compose folder in the provisioned result.
ADD requirements.txt requirements.txt will then either add the copied-version of user-specified file in nvflare_compose folder or add an empty requirements.txt.

Reply:
Ahh, missed that.  The pip install -r of the empty file just returns.  Perfect. :)